sim/src/simulator.rs

```diff
@@ -111,7 +111,9 @@ impl Simulator {
         // tick all elements to generate Rpcs
         for (i, element) in self.elements.iter_mut() {
             let rpcs = element.tick(tick);
-            self.rpc_buffer.insert(i.to_string(), rpcs);
```

Comment:
The entire tick function is in really bad shape. We should clean it up sometime.

Reply:
I'll put a TODO